lib/Sema/InstrumenterSupport.cpp

```diff
@@ -142,3 +142,18 @@ Expr *InstrumenterBase::buildIDArgumentExpr(std::optional<DeclNameRef> name,
   return new (Context) UnresolvedDeclRefExpr(*name, DeclRefKind::Ordinary,
                                              DeclNameLoc(SR.End));
 }
+
+BraceStmt *InstrumenterBase::transformBraceStmtWithLocalDeclContext(
+    BraceStmt *BS, DeclContext *localDC, const ParameterList *PL,
+    bool TopLevel) {
```

Comment:
Instead of doing this you can use RAII `llvm::SaveAndRestore` in a few spots were you need it.

Reply:
Oh cool, that looks simpler, thanks! Will take a look.

Reply:
Updated to use `llvm::SaveAndRestore`, which simplified the changes. Thanks!